components/builder-api/src/http/mod.rs

```diff
@@ -41,21 +39,27 @@ const HTTP_THREAD_COUNT: usize = 128;
 
 /// Create a new `iron::Chain` containing a Router and it's required middleware
 pub fn router(config: Arc<Config>) -> Result<Chain> {
-    let github = GitHubClient::new(&*config);
-
+    let basic = Authenticated::default();
+    let bldr = Authenticated::default().require(privilege::BUILDER);
     let router = router!(
-        get "/status" => move |r: &mut Request| status(r),
-        get "/authenticate/:code" => move |r: &mut Request| session_create(r, &github),
+        get "/status" => status,
+        get "/authenticate/:code" => session_create,
 
-        post "/jobs" => move |r: &mut Request| job_create(r),
-        get "/jobs/:id" => move |r: &mut Request| job_show(r),
+        post "/jobs" => XHandler::new(job_create).before(bldr),
```

Comment:
Nice payoff, makes things a lot clearer in my mind what privileges are required for each route. Probably should have skipped here first before digesting the admin component. No matter